### PR TITLE
Fix ueberzug backend's y position being off by one

### PR DESCRIPTION
**Pull request by Vonr**
Before:
image
After:
image

### CODE REVIEW

**Comment by 3rd** [comment]
@Vonr Hey, thank you so much for the PR!
I don't think it's a ueberzug-specific issue, it's probably something related to accounting for user-defined options around here: https://github.com/3rd/image.nvim/blob/847f5d6d37f0607a29485944768ff3ff6a6b1952/lua/image/renderer.lua#L5
Been seeing the same issue on https://github.com/3rd/image.nvim/issues/5

**Comment by Vonr** [comment]
That's interesting... So far all the images I've tested with work with this fix but ill look into renderer.lua.

**Comment by Vonr** [comment]
Might be line 109?

**Comment by Vonr** [comment]
```lua
local get_global_offsets = function()
  local x = 0
  local y = 0
  if vim.opt.number then x = x + math.max(vim.opt.numberwidth:get(), ('' .. vim.api.nvim_buf_line_count(0)):len() + (vim.opt.relativenumber:get() and 1 or 0)) end
  -- if vim.opt.signcolumn:get() ~= "no" then x = x + 2 end
  if vim.opt.showtabline == 2 then y = y + 1 end
  if vim.opt.winbar ~= "none" then y = y + 1 end
  return { x = x, y = y }
end
```
This fixes my left-right alignment but it probably breaks signcolumn alignment.
I commented that line out because I use signcolumn 'number'.

**Comment by Vonr** [comment]
```lua
local get_global_offsets = function()
  local x = 0
  local y = 0

  local numberwidth = math.max(
      vim.opt.numberwidth:get(),
      ('' .. vim.api.nvim_buf_line_count(0)):len()
        + (vim.opt.relativenumber:get() and 1 or 0))

  if vim.opt.number then x = x + numberwidth end

  local signcolumn = vim.opt.signcolumn:get()
  if signcolumn ~= "no" then
      x = x + math.max(0, 2 - (signcolumn == "number" and numberwidth or 0))
  end

  if vim.opt.showtabline == 2 then y = y + 1 end
  if vim.opt.winbar ~= "none" then y = y + 1 end
  return { x = x, y = y }
end
```
This has a pretty decent chance of working for most setups, but I honestly have no idea.

**Comment by 3rd** [comment]
@Vonr that looks great, I'll test it out today and also add some unit tests for this, and we can merge it soon.
Thanks a lot!

File: lua/image/renderer.lua [hunk 1]
```diff
@@ -5,8 +5,19 @@ local magick = require("image/magick")
 local get_global_offsets = function()
   local x = 0
   local y = 0
-  if vim.opt.number then x = x + vim.opt.numberwidth:get() end
-  if vim.opt.signcolumn ~= "no" then x = x + 2 end
+
+  local numberwidth = math.max(
+      vim.opt.numberwidth:get(),
+      ('' .. vim.api.nvim_buf_line_count(0)):len()
```
Comment (3rd) [inline lua/image/renderer.lua:11]: I think these are buffer-relative and this version applies the offsets for the active buffers to all the visible windows, looking into making this generic enough.

File: lua/image/renderer.lua [hunk 2]
```diff
@@ -106,7 +117,7 @@ local render = function(image)
     -- global offsets
     local global_offsets = get_global_offsets()
     x_offset = global_offsets.x - window.scroll_x
-    y_offset = global_offsets.y + 1 - window.scroll_y
+    y_offset = global_offsets.y - window.scroll_y
```
Comment (3rd) [inline lua/image/renderer.lua:120]: Great catch, didn't see it because the winbar logic above is wrong.

**Comment by 3rd** [comment]
@Vonr think we can rely on textoff for the gutter math, can you test if it works ok for you with the latest changes?
https://github.com/3rd/image.nvim/commit/c9576783d2eac3594fc6d264001e10bec60444b9

**Comment by Vonr** [comment]
Looks like it works, thank you!

**Pull request closed without merging**